src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java

```diff
@@ -43,29 +49,230 @@ public String key() {
     super.identify(identify);
+
+    String userId = identify.userId();
+
+    if (isNullOrEmpty(userId)) {
+      intercom.registerUnidentifiedUser();
+      logger.verbose("Intercom.client().registerUnidentifiedUser()");
+    } else {
+      Registration registration = Registration.create().withUserId(userId);
+      intercom.registerIdentifiedUser(registration);
+      logger.verbose("Intercom.client().registerIdentifiedUser(registration)");
+    }
+
+    Traits traits = identify.traits();
+    if (!isNullOrEmpty(traits)) {
+      setUserAttributes(traits);
+    }
   }
 
   @Override
   public void track(TrackPayload track) {
     super.track(track);
+
+    String eventName = track.event();
+    Properties properties = new Properties();
+    properties.putAll(track.properties());
+    Map<String, Object> price = new HashMap<>();
+    Map<String, Object> propertiesMap = new HashMap<>();
+
+    if (!isNullOrEmpty(properties)) {
+
+      Object revenueValue = properties.get(REVENUE);
+      Object totalValue = properties.get(TOTAL);
+      double amountDouble;
+      double amountInCents;
+
+      if ((revenueValue != null) && (revenueValue instanceof Double)) {
+        amountDouble = (double) properties.get(REVENUE);
+        amountInCents = amountDouble * 100;
+        price.put(AMOUNT, amountInCents);
+        properties.remove(REVENUE);
+      }
+      if ((totalValue != null) && (totalValue instanceof Double) && (revenueValue == null)) {
+        amountDouble = (double) properties.get(TOTAL);
+        amountInCents = amountDouble * 100;
+        price.put(AMOUNT, amountInCents);
+        properties.remove(TOTAL);
+      }
+      if (properties.get(CURRENCY) != null) {
+        price.put(CURRENCY, String.valueOf(properties.get(CURRENCY)));
+        properties.remove(CURRENCY);
+      }
+
+      if (!isNullOrEmpty(price)) {
+        propertiesMap.put(PRICE, price);
+      }
+
+      for (Map.Entry<String, Object> entry : properties.entrySet()) {
+        String trait = entry.getKey();
+        Object value = entry.getValue();
+        propertiesMap.put(trait, value);
+      }
+    }
+    intercom.logEvent(eventName, propertiesMap);
+    logger.verbose("Intercom.client().logEvent(%s, %s)", eventName, propertiesMap);
   }
 
   @Override
   public void group(GroupPayload group) {
     super.group(group);
+
+    if (isNullOrEmpty(group.userId())) return;
+
+    UserAttributes.Builder userAttributes = new UserAttributes.Builder();
+    Traits traits = new Traits();
+    traits.putAll(group.traits());
+    traits.put("id", group.groupId());
+    Company company = setCompany(traits);
+    userAttributes.withCompany(company);
+    intercom.updateUser(userAttributes.build());
+    logger.verbose("Intercom.client().updateUser(%s)", userAttributes);
+  }
+
+  public void reset() {
+    intercom.reset();
+    logger.verbose("Intercom.client().reset()");
+  }
+
+  private void setUserAttributes(Traits realTraits) {
```

Comment:
May not be the same in Android but iOS Intercom requires each value must be of type NSString, NSNumber or NSNull.